telegram/_forumtopic.py

```diff
@@ -0,0 +1,146 @@
+        name (:obj:`str`): Name of the topic
+        icon_color (:obj:`int`): Color of the topic icon in RGB format
+        icon_custom_emoji_id (:obj:`str`, optional): Unique identifier of the custom emoji shown
+            as the topic icon.
+
+    Attributes:
+        message_thread_id (:obj:`int`): Unique identifier of the forum topic
+        name (:obj:`str`): Name of the topic
+        icon_color (:obj:`int`): Color of the topic icon in RGB format
+        icon_custom_emoji_id (:obj:`str`, optional): Unique identifier of the custom emoji shown
+            as the topic icon.
+    """
+
+    __slots__ = ("message_thread_id", "name", "icon_color", "icon_custom_emoji_id")
+
+    def __init__(
+        self,
+        message_thread_id: int,
+        name: str,
+        icon_color: int,
+        icon_custom_emoji_id: str = None,
+        *,
+        api_kwargs: JSONDict = None,
+    ):
+        super().__init__(api_kwargs=api_kwargs)
+        self.message_thread_id = message_thread_id
+        self.name = name
+        self.icon_color = icon_color
+        self.icon_custom_emoji_id = icon_custom_emoji_id
+
+        self._id_attrs = (self.message_thread_id, self.name)
+
+
+class ForumTopicCreated(TelegramObject):
+    """
+    This object represents the content of a service message about a new forum topic created in
+    the chat.
+
+    # TODO actually maybe even just a name should be enough?
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`name` and :attr:`icon_color` are equal.
+
+    .. versionadded:: 20.0
+
+    Args:
+        name (:obj:`str`): Name of the topic
+        icon_color (:obj:`int`): Color of the topic icon in RGB format
+        icon_custom_emoji_id (:obj:`str`, optional): Unique identifier of the custom emoji shown
+            as the topic icon.
+
+    Attributes:
+        name (:obj:`str`): Name of the topic
+        icon_color (:obj:`int`): Color of the topic icon in RGB format
+        icon_custom_emoji_id (:obj:`str`, optional): Unique identifier of the custom emoji shown
+            as the topic icon.
+    """
+
+    __slots__ = ("name", "icon_color", "icon_custom_emoji_id")
+
+    def __init__(
+        self,
+        name: str,
+        icon_color: int,
+        icon_custom_emoji_id: str = None,
+        *,
+        api_kwargs: JSONDict = None,
+    ):
+        super().__init__(api_kwargs=api_kwargs)
+        self.name = name
+        self.icon_color = icon_color
+        self.icon_custom_emoji_id = icon_custom_emoji_id
+
+        self._id_attrs = (self.name, self.icon_color)
+
+    @classmethod
+    def de_json(cls, data: Optional[JSONDict], bot: "Bot") -> Optional["ForumTopicCreated"]:
+        """See :meth:`telegram.TelegramObject.de_json`."""
+        data = cls._parse_data(data)
+
+        if not data:
+            return None
+
+        return super().de_json(data=data, bot=bot)
```

Comment:
Actually I don't really understand why a test of `de_json()` will fail in `test_forum.py` unless I override `TelegramObject`'s `de_json()` here in `ForumTopicCreated`. The overridden method doesn't seem to do anything new (unlike other classes where a `User` is attached, or a timestamp converted, or whatnot).

Reply:
here's what's happening: 

1. `assert ForumTopic.de_json({}, bot=bot) is None`
2. We arrive [here](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/_telegramobject.py#L258)
3. data is actually `{}` (not `None`), so execution continues.
4. test fails.

For now to make the test pass, just replace that empty dict to `None`.
I propose to change that condition in TGObject to `if not data:` which covers the empty dict case. @Bibo-Joshi

Reply:
Oh, right, I see the difference: in the overridden method it says `if not data:`. Thanks.

Reply:
Done